planner/core/rule_eliminate_projection.go

```diff
@@ -82,6 +82,14 @@ func doPhysicalProjectionElimination(p PhysicalPlan) PhysicalPlan {
 		p.Children()[i] = doPhysicalProjectionElimination(child)
 	}
 
+	// eliminate projection in a coprocessor task
+	tableReader, isTableReader := p.(*PhysicalTableReader)
```

Comment:
It's not supported in TiKV, so I'm afraid this code will introduce overhead and complexity. I think keeping simple is best. Do less optimization until we really meet a proformance problem.

Reply:
aggred.
add a judgemet for the store type.